packages/nextjs/pages/api/og.tsx

```diff
@@ -0,0 +1,205 @@
+import { NextRequest } from "next/server";
+import { ImageResponse } from "@vercel/og";
+import { Address, createPublicClient, http } from "viem";
+import { mainnet } from "viem/chains";
+
+export const publicClient = createPublicClient({
+  chain: mainnet,
+  transport: http(),
+});
+
+export const contractAbi = [
+  {
+    constant: true,
+    inputs: [],
+    name: "name",
+    outputs: [{ name: "", type: "string" }],
+    payable: false,
+    stateMutability: "view",
+    type: "function",
+  },
+] as const;
+
+type NetworkNames = {
+  [key: number]: string;
+};
+
+const networkNames: NetworkNames = {
+  1: "Ethereum Mainnet",
+  11155111: "Sepolia",
+  10: "Optimism",
+  137: "Polygon Mainnet",
+  80001: "Polygon Mumbai",
+  42161: "Arbitrum One",
+  100: "Gnosis Chain",
+  // ... include other network names and their chain IDs here
+};
+
+const getContractName = async (contractAddress: Address) => {
+  const data = await publicClient.readContract({
+    address: contractAddress,
+    abi: contractAbi,
+    functionName: "name",
+  });
+
+  return data;
+};
```

Comment:
Lets handle the error here, maybe return some generic contract name / empty string / undefined instead ?

Reply:
<details>

<summary>
OP DAI contract : 
</summary>

http://localhost:3000/api/og?contractAddress=0xDA10009cBd5D07dd0CeCc66161FC93D7c9000da1&network=10

![Screenshot 2024-04-25 at 7 36 04 AM](https://github.com/BuidlGuidl/abi.ninja/assets/80153681/9b1509dc-5ef5-4df4-9a7e-ffb6619706b2)


</details>

Reply:
Now, we are returning an empty string, meaning that the contracts that don't have a name func won't have names at all. I am in the favor of doing this instead of returning a generic name. What do you think?

Reply:
> I am in the favor of doing this instead of returning a generic name. What do you think?

Ohh yeah totally makes sense, just found out https://github.com/BuidlGuidl/abi.ninja/pull/90#issuecomment-2076726802 (checkout 2nd part) 

Maybe instead of returning empty string we return "Contract" as fallback how we are doing [networks].tsx page ?

Reply:
I think what @Pabl0cks suggested below makes the most sense